Marlin/src/HAL/HAL_STM32F1/persistent_store_sdcard.cpp

```diff
@@ -64,7 +64,7 @@ static char HAL_STM32F1_eeprom_content[HAL_STM32F1_EEPROM_SIZE];
     if (!card.isDetected()) return false;
 
     SdFile file, root = card.getroot();
-    if (!file.open(&root, EEPROM_FILENAME, O_CREAT | O_WRITE | O_TRUNC))
+    if (file.open(&root, EEPROM_FILENAME, O_CREAT | O_WRITE | O_TRUNC))
```

Comment:
hmm why did you change that ?

Reply:
@thinkyhead: `file.open` returns true on success, so in fact the code was correct before you changed it.

Reply:
Typo. I was going to do something else there and apparently didn't finish.

Reply:
yes, but you merged it with that, could you fix it on the git ?